e2e/example_runner.go

```diff
@@ -57,19 +57,21 @@ func (f *ExamplesFramework) Create(fileName string) {
 	// try to serialize it into Unstructured or UnstructuredList
 	data, err := ioutil.ReadFile(fileName)
 	Expect(err).NotTo(HaveOccurred())
-	jsonData, err := utilyaml.ToJSON(data)
-	Expect(err).NotTo(HaveOccurred())
-	var ust runtime.Unstructured
-	err = runtime.DecodeInto(api.Codecs.UniversalDecoder(), jsonData, &ust)
-	if err != nil {
-		var ustList runtime.UnstructuredList
-		err = runtime.DecodeInto(api.Codecs.UniversalDecoder(), jsonData, &ustList)
+
+	utils.ForEachYamlDocument(data, func(doc []byte) {
```

Comment:
Hmmm... I wander if we could use there `api.codecs.UniversialDeserializer()` with `"application/yaml"` as media type for `runtime.SerializerInfoForMediaType(...)` (as in `TestUniversalDeserializer` in `serialization_proto_test.go` in k8s sources) then create `streaming.NewDecoder` using `framer` from gathered earlier serializer info (as in `TestObjectWatchFraming` in `serialization_test.go` in k8s sources), and then simply looping in for over `Decode(nil, object)` waiting to `EOF`. That way we also don't need to read whole file at start, we can pass opened file to frame reader (it will close it at `EOF` afaik).

This would be IMO a bit more k8s way to handle streamed marshaled data.

I have no strong opinion on that - your manual splitting with regex does almost the same and is ok for me (it's clean/simple), but uses "own" method, which also would require an unit test...